### PR TITLE
fix(codegen): modify generator scripts after separation of bazel rules

### PR DESCRIPTION
**Pull request by emmileaf**
**This PR:** edits generator scripts given the separation of the `java_gapic_spring_library` rule in https://github.com/googleapis/gapic-generator-java/pull/1065 (please see its description for context of edits).

The script now makes local edits to:
- `googleapis/repository_rules.bzl` (add switch)
- `google/cloud/<client_lib_name>/v1/BUILD.bazel` (add rule name to load, and rule definition using similar arguments as `java_gapic_library`)

**Note for review:** the initial goal of these changes is to have something to unblock the code generation workflow, but there's currently a lot of perl mess so any suggestions on cleaner/more reliable file processing would be welcomed!

### CODE REVIEW

**Review by zhumin8 — COMMENTED** [review]
I agree on unblocking the generation workflow. So we could probably merge this in while working on improvements.
On making it more robust, I found this [buildozer](https://github.com/bazelbuild/buildtools/blob/master/buildozer/README.md) tool that looks neat. E.g. these two commands can achieve most of what we want building the `java_gapic_spring_library` rule.
```
# Add a new rule at the end of the file
buildozer 'new java_library foo' //pkg:__pkg__

# Copy an attribute from `protolib` to `py_protolib`.
buildozer 'copy testonly protolib' //pkg:py_protolib
```
Drawback is that it has dependencies on `Go`. But GithubAction does have https://github.com/actions/setup-go, so I guess it's not that bad.

**Comment by emmileaf** [comment]
> On making it more robust, I found this [buildozer](https://github.com/bazelbuild/buildtools/blob/master/buildozer/README.md) tool that looks neat. E.g. these two commands can achieve most of what we want building the `java_gapic_spring_library` rule.

Oh buildozer does look like a better tool for handling BUILD.bazel file changes (esp. copying over attributes) - thanks for the suggestion @zhumin8 ! I’ll look into how to bring it in for our workflow, and mark this PR as ready for review in the meanwhile.

File: generator/generate-one.sh [hunk 1]
```diff
@@ -39,20 +39,36 @@ if [[ $download_repos -eq 1 ]]; then
...
+SPRING_RULE_NAME="    \\\"java_gapic_spring_library\\\","
+perl -0777 -pi -e "s/(load\((.*?)\"java_gapic_library\",)/\$1\n$SPRING_RULE_NAME/s" google/cloud/$client_lib_name/v1/BUILD.bazel
+# Duplicate java_gapic_library rule definition
+perl -0777 -pi -e "s/(java_gapic_library\((.*?)\))/\$1\n\n\$1/s" google/cloud/$client_lib_name/v1/BUILD.bazel
+# Update rule name to java_apic_spring_library
+perl -0777 -pi -e "s/(java_gapic_library\()/java_gapic_spring_library\(/s" google/cloud/$client_lib_name/v1/BUILD.bazel
+# Update name argument to have _spring appended
+perl -0777 -pi -e "s/(java_gapic_spring_library\((.*?)name = \"(.*?)\")/java_gapic_spring_library\(\$2name = \"\$3_spring\"/s" google/cloud/$client_lib_name/v1/BUILD.bazel
+# todo: better way to remove the following unused arguments?
+perl -0777 -pi -e "s/(java_gapic_spring_library\((.*?)(\n    test_deps = \[(.*?)\],))/java_gapic_spring_library\(\$2/s" google/cloud/language/v1/BUILD.bazel
```
Comment (zhumin8) [inline generator/generate-one.sh:55]: `language` here should be `$client_lib_name`
```suggestion
perl -0777 -pi -e "s/(java_gapic_spring_library\((.*?)(\n    test_deps = \[(.*?)\],))/java_gapic_spring_library\(\$2/s" google/cloud/$client_lib_name/v1/BUILD.bazel
```

Comment (emmileaf) [reply generator/generate-one.sh]: Thanks for catching this! Updated.

**Review by diegomarquezp — APPROVED** [review]
LGTM

**Pull request merged**